pkg/nodeconfig/nodeconfig.go

```diff
@@ -267,6 +250,22 @@ func (nc *nodeConfig) Configure() error {
 		if err := nc.applyLabelsAndAnnotations(nil, map[string]string{metadata.VersionAnnotation: version.Get()}); err != nil {
 			return errors.Wrapf(err, "error updating version annotation on node %s", nc.node.GetName())
 		}
+		// If we deploy on Azure with CCM support, we have to explicitly remove the cloud taint, because cloud node
+		// manager running on the node can't do it itself, due to lack of RBAC permissions given by the node
+		// kubeconfig it uses.
+		if ownedByCCM && nc.platformType == configv1.AzurePlatformType {
```

Comment:
- Is this how it is done on the Linux side?
- Shouldn't this be done before applying `nc.applyLabelsAndAnnotations` because at that point we still have not finished configuring and this is now WMCO's responsibility?

Reply:
>On linux cloud-node-manager removes the taint itself, which it can do as it runs in a pod.

My concern is we could be removing the taint while the `cloud-node-manager` has a failure and we really shouldn't be removing the taint. IOW, we seem to be removing the taint without any feedback from the `cloud-node-manager`.

Reply:
That definitely could be an issue. That behavior still exists before this PR, in that as long as the service starts successfully, we have no idea if it is doing the correct thing before we remove the annotation. The only real solution to this is to run it as a pod, as it is meant to be.